src/Xmf2.Components.Droid/Controls/ChipCloud/ChipCloudAdapter.cs

```diff
@@ -71,7 +71,12 @@ private void UpdateComponentDictionary(List<Remove> removeList, List<Move> moveL
 			foreach (var move in moveList.OrderBy(x => x.OldPos))
 			{
 				_componentDictionary[move.NewPos] = oldDic[move.OldPos];
-				_componentDictionary.Remove(move.OldPos);
+
+				if (removeList.Count > 0 && !moveList.Any(x => x.NewPos == move.OldPos))
+				{
+					//CLA 17/05/2023 : remove element from dictionary only if an element needs to be removed and there is no inversion between this element and another
```

Comment:
//CLA 17/05/2023 : remove element from dictionary only if an element has been removed (tail must be truncated by moving all next elements)
                                         // Futhermore, we need to check if there is no inversion between this element and another (in this case dictionary must keep elements)